solarkraft/src/fetcher/callDecoder.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * @license
+ * [Apache-2.0](https://github.com/freespek/solarkraft/blob/main/LICENSE)
+ */
+
+/**
+ * The code to extract Stellar operations that correspond to Soroban contract calls.
+ *
+ * Igor Konnov, 2024.
+ */
+
+import sdk, { Address } from '@stellar/stellar-sdk'
+import { ContractCallEntry, FieldsMap } from './storage.js'
+import { Maybe, just, none } from '@sweet-monads/maybe'
+import { OrderedMap } from 'immutable'
+
+/**
+ * Decode a contract call from a Horizon operation.
+ * @param op an operation
+ * @param matcher a quick matcher over the contractId to avoid expensive deserialization
+ */
+export async function extractContractCall(
+    op: any,
+    matcher: (contractId: string) => boolean
+): Promise<Maybe<ContractCallEntry>> {
+    // https://developers.stellar.org/network/horizon/api-reference/resources/operations/object
+    if (op.function === 'HostFunctionTypeHostFunctionTypeInvokeContract') {
+        // This operation represents a smart contract call.
```

Comment:
Can't you just return `none` if `!==`.
Bit hard to follow controlflow here with the indentation.